semgrep/semgrep/version.py

```diff
@@ -0,0 +1,118 @@
+import os
+import time
+from pathlib import Path
+from typing import Optional
+
+import requests
+from packaging.version import InvalidVersion
+from packaging.version import Version
+
+from semgrep import constants
+from semgrep import util
+
+VERSION_CHECK_URL = str(
+    os.environ.get(
+        "SEMGREP_VERSION_CHECK_URL", "https://semgrep.live/api/check-version"
+    )
+)
+VERSION_CHECK_TIMEOUT = int(
+    os.environ.get(
+        "SEMGREP_VERSION_CHECK_TIMEOUT", 2  # Don't block user's for too long
+    )
+)
+VERSION_CACHE_PATH = Path(
+    os.environ.get(
+        "SEMGREP_VERSION_CACHE_PATH",
+        Path(os.path.expanduser("~/.cache")) / "semgrep_version",
```

Comment:
Using `.cache` isn't ideal on OSX, but I figured this was fine compared to pulling in another dependency like [`appdirs`](https://pypi.org/project/appdirs/) just to grab a directory. If a user has strong feels about this they can always set `SEMGREP_VERSION_CACHE_PATH`.